pkg/tfbridge/provider.go

```diff
@@ -1235,14 +1235,22 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 		return arr
 	}
 
-	// Ensure that outputs are deterministic to enable gRPC testing.
-	sort.Strings(replaces)
-	sort.Strings(stables)
+	sorted := func(arr []string) []string {
+		sort.Strings(arr)
+		return arr
+	}
+
+	if a, ok := diff.(interface {
+		Attributes() map[string]shim.ResourceAttrDiff
```

Comment:
Why is the cast here needed? `diff` is already a `shim.InstanceDiff` which should have that method

Reply:
It doesn't:

https://github.com/pulumi/pulumi-terraform-bridge/blob/aa7015a81afda1fe211264f081bbdfa181f5d75b/pkg/tfshim/shim.go#L47-L53

It has `Attribute`, but we want the set here.